service-registry/src/main/java/org/apache/servicecomb/serviceregistry/task/MicroserviceRegisterTask.java

```diff
@@ -261,6 +263,24 @@ private boolean compareAndReRegisterSchema(Entry<String, String> localSchemaEntr
       String scSchemaContent = srClient.getSchema(microservice.getServiceId(), scSchema.getSchemaId());
       String localSchemaContent = localSchemaEntry.getValue();
 
+      if (!StringUtils.isEmpty(scSchemaContent) && !StringUtils.isEmpty(localSchemaContent)) {
+        Swagger scSwagger = SwaggerUtils.parseSwagger(scSchemaContent);
+        Swagger localSwagger = SwaggerUtils.parseSwagger(localSchemaContent);
+        if (scSwagger.equals(localSwagger)) {
+          return true;
+        }
+        if (ServiceRegistryConfig.INSTANCE.isIgnoreSwaggerDifference()) {
```

Comment:
put `if (ServiceRegistryConfig.INSTANCE.isIgnoreSwaggerDifference()) {` outside of 

 if (!StringUtils.isEmpty(scSchemaContent) && !StringUtils.isEmpty(localSchemaContent)) {

block.

Reply:
done. I have put it behind EQUALS block.